base/collection_common.go

```diff
@@ -32,3 +32,19 @@ func (s ScopeAndCollectionName) ScopeName() string {
 func (s ScopeAndCollectionName) CollectionName() string {
 	return s.Collection
 }
+
+// String returns a dot-separated formatted scope and collection name.
+func (s ScopeAndCollectionName) String() string {
+	return s.Scope + ScopeCollectionSeparator + s.Collection
+}
+
+type ScopeAndCollectionNames []ScopeAndCollectionName
+
+// ScopeAndCollectionNames returns a dot-separated formatted slice of scope and collection names.
+func (s ScopeAndCollectionNames) ScopeAndCollectionNames() []string {
```

Comment:
This would be useful https://github.com/couchbase/sync_gateway/blob/master/rest/utilities_testing.go#L2320

Reply:
Best I can do is this to avoid unnecessary double-iteration.

```go
collections = append(collections, base.ScopeAndCollectionName{
	Scope:      collection.ScopeName,
	Collection: collection.Name,
}.String())
```